Bottom: f6db665131989166508d7508428282697dffd14a
Top:    400e803f70e229fc3f32cf84ff318fe6fc55b1a7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 02:04:25 +0000

configure.ac: Use CPPFLAGS, not CFLAGS for our include directives.

We were previously using CFLAGS for adding pre-processor includes, where
these should really end up in the CPPFLAGS variable. Confusingly, their
constituents tend to come from variables set by autoconf / pkgconfig
expansions in the form of LIBNAME_CFLAGS.

Doing this allows the user to easily override CFLAGS at build time without
needing to replicate the enture include path list.


---

diff --git a/configure.ac b/configure.ac
index 7e8348b..39ae394 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1105,9 +1105,9 @@ AC_DEFINE_UNQUOTED([COORD_TYPE],[$COORD_TYPE],
 AC_DEFINE_UNQUOTED([COORD_MAX],[$COORD_MAX],
   [Maximum value of coordinate type])
 
-# ------------- Complete set of CFLAGS and LIBS -------------------
+# ------------- Complete set of CPPFLAGS and LIBS -------------------
 
-CFLAGS="$X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $GD_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS $GLU_CFLAGS $GL_CFLAGS"
+CPPFLAGS="$X_CFLAGS $DBUS_CFLAGS $GLIB_CFLAGS $GTK_CFLAGS $GD_CFLAGS $CAIRO_CFLAGS $GTKGLEXT_CFLAGS $GLU_CFLAGS $GL_CFLAGS"
 LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GLIB_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS $CAIRO_LIBS $GTKGLEXT_LIBS $GLU_LIBS $GL_LIBS"
